core-concepts/modules/ROOT/pages/typeql/sql-vs-typeql.adoc

```diff
@@ -0,0 +1,718 @@
+
+==== INNER JOIN (or JOIN)
+
+An INNER JOIN returns only rows where there is a match in both tables. In TypeQL, this is the default behavior when matching relations.
+
+**SQL:**
+[source,sql]
+----
+SELECT p.name, e.id
+FROM person p
+INNER JOIN employment e ON p.id = e.employee_id;
+----
+
+**TypeQL:**
+[,typeql]
+----
+#!test[read, count=1]
+match
+  $p isa person, has name $pname;
+  $e isa employment, links (employee: $p);
+select $pname, $e;
+----
+
+This returns only persons who have an employment relation.
+
+==== LEFT JOIN
+
+A LEFT JOIN returns all rows from the left table (person), and matching rows from the right table (employment).
+If there's no match, NULL values are returned for the right table columns. In TypeQL, use the `try` construct for optional matching.
+
+**SQL:**
+[source,sql]
+----
+SELECT p.name, e.id
+FROM person p
+LEFT JOIN employment e ON p.id = e.employee_id;
+----
+
+**TypeQL:**
+[,typeql]
+----
+#!test[read, count=3]
+match
+  $p isa person, has name $pname;
+  try {
+    $e isa employment, links (employee: $p);
+  };
+select $pname, $e;
+----
+
+This returns all persons, with their employment if they have one.
+Persons without employment will still appear in the results (with `$e` unbound).
+
+==== RIGHT JOIN
+
+A RIGHT JOIN returns all rows from the right table (employment), and matching rows from the left table (person).
+If there's no match, NULL values are returned for the left table columns. In TypeQL, reverse the pattern to start from the right side.
+
+**SQL:**
+[source,sql]
+----
+SELECT p.name, e.id
+FROM person p
+RIGHT JOIN employment e ON p.id = e.employee_id;
+----
+
+**TypeQL:**
+[,typeql]
+----
+#!test[read, count=1]
+match
+  $e isa employment;
+  try {
+    $p isa person, has name $pname;
+    $e links (employee: $p);
+  };
+select $pname;
+----
+
+This returns all employments, with employee names if they exist.
+Employments without matching persons will still appear in the results (with `$pname` unbound).
+
+==== CROSS JOIN
+
+A CROSS JOIN returns the Cartesian product of two tables - every row from the first table combined with every row from the second table.
+In TypeQL, match entities from both types independently without requiring any relation between them.
+
+**SQL:**
+[source,sql]
+----
+SELECT p.name, e.id
+FROM person p
+CROSS JOIN employment e;
+----
+
+**TypeQL:**
+[,typeql]
+----
+#!test[read, count=3]
+match
+  $p isa person, has name $pname;
+  $e isa employment;
+select $pname, $e;
+----
+
+This returns every combination of person and employment, regardless of whether they have any relationship.
+The result set size is the product of the number of persons and employments.
+
+==== FULL OUTER JOIN
+
+A FULL OUTER JOIN returns all rows from both tables, with NULL values where there's no match.
+
+TypeQL doesn't yet have an efficient equivalent of full outer joins! Please leave us a message if this operation is required for your application.
```

Comment:
This reads as a bit defeatist right now. The first sentence could be closer to "Full joins do not translate well to TypeQL. If your application requires full joins, ..."